src/org/exist/dom/persistent/ElementImpl.java

```diff
@@ -1427,16 +1444,16 @@ public Node removeChild(final Txn transaction, final Node oldChild) throws DOMEx
 
         final NodePath oldPath = oldNode.getPath();
         try(final DBBroker broker = ownerDocument.getBrokerPool().getBroker()) {
-            //May help getReindexRoot() to make some useful things
-            broker.getIndexController().setDocument(ownerDocument);
-            final IStoredNode reindexRoot = broker.getIndexController().getReindexRoot(oldNode, oldPath, false);
-            broker.getIndexController().setMode(ReindexMode.REMOVE_SOME_NODES);
+            IndexController indexes = broker.getIndexController();
+
```

Comment:
```
listener = null;
indexes.reindex(transaction, reindexRoot, ReindexMode.REMOVE_SOME_NODES);
```

`indexes.reindex` have it owen call of `IndexController.setDocument` here
`StreamListener stream = getStreamListener(node.getOwnerDocument(), mode);`